pysoa/test/stub_service.py

```diff
@@ -301,165 +312,178 @@ def __init__(self, service, action, body=None, errors=None):
         self.attribute_name = None
         self.new = mock.DEFAULT
 
+        self._stub_action_responses_outstanding = defaultdict(dict)
+        self._stub_action_responses_to_merge = defaultdict(dict)
+
     def __enter__(self):
-        self._wrapped_client_call_actions_method = Client.call_actions
-        self._wrapped_client_call_jobs_parallel = Client.call_jobs_parallel
+        self._wrapped_client_send_request = Client.send_request
+        self._wrapped_client_get_all_responses = Client.get_all_responses
+        self._services_with_calls_sent_to_wrapped_client = set()
 
         mock_action = self._MockAction(name='{}.{}'.format(self.service, self.action))
 
         if self.body or self.errors:
-            mock_client = StubClient()
-            mock_client.stub_action(self.service, self.action, body=self.body, errors=self.errors)
-            mock_action.side_effect = lambda body: self._wrapped_client_call_actions_method(
-                mock_client,
-                self.service,
-                [ActionRequest(self.action, body=body)],
-                raise_action_errors=False,
+            mock_action.side_effect = lambda _: ActionResponse(self.action, errors=self.errors, body=self.body)
+
+        @wraps(Client.send_request)
+        def wrapped_send_request(client, service_name, actions, *args, **kwargs):
+            assert isinstance(service_name, six.text_type), 'Called service name "{}" must be unicode'.format(
+                service_name,
             )
 
-        @wraps(Client.call_jobs_parallel)
-        def wrapped_call_jobs_parallel(client, jobs, *args, **kwargs):
-            jobs_to_reinsert = set()
-            actions_to_reinsert = OrderedDict()
-            for j_i, job_request in enumerate(jobs):
-                service_name, actions = job_request['service_name'], job_request['actions']
-                assert isinstance(service_name, six.text_type), 'Called service name "{}" must be unicode'.format(
+            actions_to_send_to_mock = OrderedDict()
+            actions_to_send_to_wrapped_client = []
+            for i, action_request in enumerate(actions):
```

Comment:
nitt: single letter var name.

Reply:
My understanding is that this is considered okay when the single-letter is used for the enumerated index of an iterating object.